mp/src/game/client/hud_chat.cpp

```diff
@@ -162,6 +172,29 @@ void CHudChat::MsgFunc_LobbyUpdateMsg(bf_read &msg)
         SpectatorUpdate(personID, k_steamIDNil);
 }
 
+void CHudChat::GetTimestamp(char *pBuffer, int maxLen)
+{
+    time_t now = time(nullptr);
+    struct tm *tm = localtime(&now);
+    Q_snprintf(pBuffer, maxLen, "%c%s[%02d:%02d] ", COLOR_HEXCODE, "CCCCCC", tm->tm_hour, tm->tm_min);
+}
+
+void CHudChat::Printf(int iFilter, const char *fmt, ...)
+{
+    va_list marker;
+    char msg[4096];
+
+    va_start(marker, fmt);
+    Q_vsnprintf(msg, sizeof(msg), fmt, marker);
+    va_end(marker);
+
+    char timestamp[16] = "";
+    if (mom_chat_timestamps_enable.GetBool())
```

Comment:
Why not move these two lines into the GetTimestamp method? You can early return with copying the empty string into the char buffer if the convar isn't enabled.

I only bring this up because this snippet seems to be repeated earlier in the file as well.